python/arcticdb/diagnostics_engine/library_utils.py

```diff
@@ -0,0 +1,40 @@
+from arcticdb.version_store import NativeVersionStore
+from IPython.display import display, Markdown
+import pandas as pd
+
+
+def check_and_adapt_library(lib):
+    """
+    Verifies that the type of the library is compatible with the diagnostics engine, and if necessary,
+    adapts the library to the type used internally by the engine
+    """
+    if isinstance(lib, NativeVersionStore):
+        return lib
+    else:
+        display(Markdown(f"the provided library is not supported"))
+        return None
+
+
+def get_string_version(as_of):
+    if as_of is not None:
+        if isinstance(as_of, pd.Timestamp):
```

Comment:
We support some other time types as well:
https://github.com/man-group/ArcticDB/blob/master/python/arcticdb/supported_types.py#L21

Reply:
fine